File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.swift
func solution(_ n:Int) -> Int {
    var table = Array(repeating: 0, count: n + 1)
    
    if n == 1 {
        return 1
    } else if n == 2 {
        return 2
    }

    (table[1], table[2]) = (1, 2)
    
    for i in 3...n {
        table[i] = (table[i - 1] + table[i - 2]) % 1234567
    }

    return table[n]
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    if n == 1 or n == 2:
        return n

    memo = [0 for _ in range(n)]
    (memo[0], memo[1]) = (1, 2)
    
    for i in range(2, n):
        memo[i] = (memo[i-1] + memo[i-2]) % 1234567
    
    return memo[n-1]